### PR TITLE
Fix:Post Parent-child category editor issues

### PR DESCRIPTION
**Pull request by supriyajjadhav**
[Trello Task](https://trello.com/c/V1UT94nB/136-uag-posts-block-to-have-a-selection-for-related-posts)
### Description
Resolve parent-child category posts issue in the editor.

### Types of changes
- Added function to check the parent-child category relationship in the taxonomy list to display posts in different layouts (Grid, Masonry, Carousel).
- Removed Select option in taxonomyListOptions.

### Test Cases
- Create parent-child categories.
- Add categories to the posts.
- In the Editor Please select any parent category.
- Check It will display all the related posts.

### How has this been tested?
Tested whether it is displaying all the related posts in the editor.

### Checklist:
- [x] My code is tested
- [x] My code follows accessibility standards 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: src/blocks/post/post-grid/edit.js [hunk 1]
```diff
@@ -1149,8 +1148,26 @@ export default withSelect( ( select, props ) => {
...
+
+	var category = [];
+	if( undefined !== categories || null !== categories || '' !== categories){		
+		var temp = parseInt(categories);
+		category.push(temp);
+		CheckCategories(categoriesList);
 	
-	latestPostsQuery[rest_base] = categories
+	}
+	function CheckCategories(catlist){
```
Comment (RajkiranBagal) [inline src/blocks/post/post-grid/edit.js:1159]: Let's perform all these operations in PHP itself as discussed on call as this loop gets called multiple time & is not good for performance.

Comment (supriyajjadhav) [reply src/blocks/post/post-grid/edit.js]: Done

**Review by RajkiranBagal — APPROVED** [review]

File: readme.txt [hunk 2]
```diff
@@ -178,6 +178,7 @@ The awesome React FontIconPicker is developed by [Alessandro Benoit](http://code
 * Fix: FAQ-Schema - FAQ console error when accessing child by hashval.
 * Fix: Taxonomy List - Fixed issue with broken links.
 * Fix: Testinomial - Fixed carousel not being displayed on RTL websites.
+* Fix: Post - Fixed related posts issue in editor mode in Post Grid, Post Masonry, Post Carousel and Post Timeline blocks.
```
Comment (vrundakansara) [inline readme.txt:181]: `* Fix: Post Layout - Child Posts not visible when parent category is selected in editor mode.`

Comment (supriyajjadhav) [reply readme.txt]: Done

**Review by vrundakansara — APPROVED** [review]

**Pull request merged**